genmc-sys/cpp/src/MiriInterface/EventHandling.cpp

```diff
@@ -255,7 +259,7 @@ auto MiriGenmcShim::handle_malloc(ThreadId thread_id, uint64_t size, uint64_t al
 
 auto MiriGenmcShim::handle_free(ThreadId thread_id, uint64_t address) -> bool {
     const auto pos = inc_pos(thread_id);
-    GenMCDriver::handleFree(pos, SAddr(address), EventDeps());
+    GenMCDriver::handleFree(nullptr, pos, SAddr(address), EventDeps());
```

Comment:
`handleFree` seems to return a `std::optional<VerificationError>` now, does that mean we can resolve the FIXME below and remove this hack?

Reply:
IIRC the problem was that `handleFree` didn't actually return an error, so we had to call `getResult` to check whether there was an error.

Yeah, let's move that to a future PR.